src/components/Tile.js

```diff
@@ -0,0 +1,50 @@
+import React, { Component } from 'react';
+import { StyleSheet, Text } from 'react-primitives';
+import PropTypes from 'prop-types';
+
+export default class Tile extends Component {
+    static propTypes = {
+        onPress: PropTypes.func.isRequired,
+        tileValue: PropTypes.number.isRequired,
+        enabled: PropTypes.bool,
+    };
+
+    static defaultProps = {
+        enabled: true,
+    };
+
+    move() {
+        if (this.props.enabled) {
+            this.props.onPress(this.props.tileValue);
+        }
+    }
+
+    render() {
+        return (
+            <Text style={styles.tile} onPress={this.move}>
```

Comment:
You're probably going to have troubles here. See https://reactjs.org/docs/handling-events.html